src/cborparser.c

```diff
@@ -108,7 +108,7 @@ static CborError preparse_value(CborValue *it)
     }
 
     size_t bytesNeeded = descriptor < Value8Bit ? 0 : (1 << (descriptor - Value8Bit));
```

Comment:
1 is int, but the value of descriptor is already range-checked. By construction, it's in the range 0-27. So bytesNeeded is either 0, 1, 2, 4, or 8. That's an int.

That int is extended to size_t, but there's no sign conversion.